### DISCUSSION TITLE
How do I turn off Visual Studio automatic formatting?

### DISCUSSION THREAD

**Discussion by ohmaya** [o3de/o3de#13353] [Help Wanted]
Projects generated for O3DE have automatic formatting on, there might be a setting in O3DE's repo. Thanks.

**Comment by AMZN-daimini** [comment]
Hey ohmaya,
I'm not sure if it's going to fix your problem, but if your project is inside the o3de folder one thing that may help is seeing if removing or clearing the `\\o3de\.editorconfig` file affects the formatting. (May need to re-generate the solution to see effects).

Otherwise, all other formatting options I've seen were usually in the VS Settings, not in the Project Configuration. But if you can find which setting in Project Configuration is affecting your project we can try to figure out where it is set.

**Reply by ohmaya** [reply]
.editorconfig does not seem to have a setting about automatic formatting.

**Comment by rgba16f** [comment]
I believe you are referring to the clang-formatting that is default turned on. You can either locally remove or edit `o3de/.clang-format` or in visual studio options you can tell it to not apply formatting in (VS2022) Tools > Options, Text Editor > c++ > Formatting > Enable Clang Format support
